### PR TITLE
Feature/task1

### PR DESCRIPTION
**Pull request by MrSkorohod**

### CODE REVIEW

File: tasks/task1.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
+/* Task 1 */
+function summ(...args) {
+  let sum = 0;
+  for (let i = 0; i < args.length; i += 1) {
```
Comment (ElizavetaSelivanova) [inline tasks/task1.js]: Good, but try to find another way to solve this task =)

File: tasks/task1.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
+/* Task 1 */
+function summ(...args) {
```
Comment (ElizavetaSelivanova) [inline tasks/task1.js:2]: Add check for existence (if undefined -> return) to avoid errors

(*for every function)

File: tasks/task1.spec.js [hunk 3]
```diff
@@ -0,0 +1,122 @@
+describe("Task 1.", () => {
```
Comment (ElizavetaSelivanova) [inline tasks/task1.spec.js:1]: Add some empty lines between logical blocks to make code more readable

**Review by ElizavetaSelivanova — CHANGES_REQUESTED** [review]

File: .eslintrc.js [hunk 4]
```diff
@@ -13,6 +13,7 @@ module.exports = {
     sourceType: 'module',
   },
   rules: {
+    "no-unused-vars": "off",
```
Comment (angelinabylitskaya) [inline .eslintrc.js]: turn on this rule

File: tasks/task1.js (hunk 2 above)
Comment (angelinabylitskaya) [reply tasks/task1.js:2]: check that value exists, not only for undefined

File: tasks/task1.js [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+function getTen() {
+  return 10;
+}
+
+function getTenString() {
+  return "10";
+}
+
+function summAdvanced(...args) {
+  if (args === undefined || args === null) return false;
```
Comment (angelinabylitskaya) [inline tasks/task1.js]: should return undefined

File: tasks/task1.js [hunk 6]
```diff
@@ -0,0 +1,109 @@
+/* Task 1 */
+function summ(...args) {
+  if (args === undefined || args === null) return false;
```
Comment (angelinabylitskaya) [inline tasks/task1.js]: should return undefined

File: tasks/task1.js [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+
+summAdvanced("abc", 1, "2", getTen, getTenString, getRandomNumber);
+
+/* Task 3 */
+function isValueExists(value) {
+  if (value === undefined || value === null) {
+    return false;
+  }
+
+  return true;
```
Comment (angelinabylitskaya) [inline tasks/task1.js]: can be simplified in one line

File: tasks/task1.js [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+}
+callWithFunctionResult(doubleValue, getFour);
+
+/* Task 5 */
+function callWhileStringIsNotEmpty(string, func) {
+  if (string === undefined || func === undefined || string === null || func === null) {
+    return false;
+  }
+
+  if (typeof string !== "string") return null;
```
Comment (angelinabylitskaya) [inline tasks/task1.js]: Different return results on invalid input

**Review by angelinabylitskaya — APPROVED** [review]

**Pull request merged**